src/uu/ls/src/ls.rs

```diff
@@ -569,16 +569,20 @@ impl Config {
         };
 
         let width = match options.value_of(options::WIDTH) {
-            Some(x) => match x.parse::<u16>() {
-                Ok(u) => {
-                    if u != 0 && x.starts_with('0') {
-                        return Err(LsError::InvalidLineWidth(x.into()).into());
-                    } else {
-                        u
+            Some(x) => {
+                if x.starts_with('0') && x.len() > 1 {
```

Comment:
Extra check for the len is not technically needed, from_str_radix with a 0 gets us correct results.
But using it we will only use from_str_radix on numbers that truly look octal (0 and one more following) and thje rest, like now, goes through the normal parse function.